superset/assets/javascripts/explorev2/exploreUtils.js

```diff
@@ -1,26 +1,40 @@
 /* eslint camelcase: 0 */
-export function getExploreUrl(form_data, endpoint = 'base', force = false) {
+import URI from 'urijs';
+
+export function getExploreUrl(form_data, endpointType = 'base', force = false, curUrl = null) {
   if (!form_data.datasource) {
     return null;
   }
+
+  let uri = URI(window.location.search);
+  // curUrl is used for tests
```

Comment:
Here I want for arbitrary url parameters in the current context (explore, dashboard and sqllab) to propagate in related urls and requests.

Now for the unit tests I need to either find a way to spoof the current URL (mimicking a user adding `?foo=bar` in the URL) somehow in Mocha, or (like I did here), make it an optional argument that is used in the context of unit tests.

This function is used a lot, and would rather not change the dozens of calls to pass a global var that's always in scope already...